### ISSUE TITLE
Use FluentValidation to handle miniLcm api validation

### ISSUE THREAD

**Issue by hahn-kev** [sillsdev/languageforge-lexbox#1236]
We need a clear and consise way to validate requests for miniLcm, a great library for this is [FluentValidation](https://docs.fluentvalidation.net/en/latest/), I'd like to see that integrated, to start with the following can be validated:
- [x] Creating an entry with a sense that has an EntryId not matching it's parent is invalid (an empty/null id is fine)
- [x] Similar for complex forms, the expected property must match it's parent entry
- [x] Update and Create APIs should reject input with a deleted date set
- [x] Validation based on the current state, EG circular references

Examples can be found here https://github.com/sillsdev/languageforge-lexbox/tree/develop/backend/FwLite/MiniLcm/Validators

For now this validation is not data integrity based, but ensuring required fields are provided that sort of thing. Validation should follow Fieldworks as closely as we can.

**Comment by rmunn** [comment]
> - [ ] Validation based on the current state, EG circular references

@hahn-kev Could you clarify this one a little?

**Comment by hahn-kev** [comment]
Yeah, for example an entry can't reference itself as a complex form, that sort of thing.

**Comment by rmunn** [comment]
So, something like this then:

```cs
    public EntryValidator()
    {
        // ...
        RuleForEach(e => e.Components).Must(NotBeSelfReference);
        RuleForEach(e => e.Components).Must(HaveCorrectEntryReference);
    }

    private bool NotBeSelfReference(Entry entry, ComplexFormComponent component)
    {
        return component.ComponentEntryId != entry.Id;
    }

    private bool HaveCorrectEntryReference(Entry entry, ComplexFormComponent component)
    {
        return component.ComplexFormEntryId == entry.Id;
    }
```

**Comment by rmunn** [comment]
With PR #1344 merged, I believe this is completed. Closing. @hahn-kev - Please reopen if there's more work to be done on this.
